sandy/fy.py

```diff
@@ -422,10 +422,22 @@ def custom_perturbation(self, zam, mt, e, zap, pert):
         >>> nfpy_pert = nfpy.custom_perturbation(922350, 459, 0.0253, 551370, 0.9)
         >>> comp = nfpy_pert.data.query('ZAM==922350 & ZAP==551370 & MT==459 & E==0.0253').squeeze().FY
         >>> assert np.setdiff1d(nfpy_pert.data.values, nfpy.data.values) == comp
+
+        >>> pert = pd.Series([0.9, 0.5], index=[541350, 551370])
+        >>> nfpy_pert = nfpy.custom_perturbation(922350, 459, 0.0253, [541350, 551370], pert)
+        >>> comp1 = nfpy_pert.data.query('ZAM==922350 & ZAP==551370 & MT==459 & E==0.0253').squeeze().FY
+        >>> f1 = nfpy.data.query('ZAM==922350 & ZAP==551370 & MT==459 & E==0.0253').squeeze().FY
+        >>> assert f1 * 0.5 == comp1
+        >>> comp2 = nfpy_pert.data.query('ZAM==922350 & ZAP==541350 & MT==459 & E==0.0253').squeeze().FY
+        >>> f2 = nfpy.data.query('ZAM==922350 & ZAP==541350 & MT==459 & E==0.0253').squeeze().FY
+        >>> assert f2 * 0.9 == comp2
         """
         df = self.data.copy()
-        mask = (df.ZAM == zam) & (df.ZAP == zap) & (df.MT == mt) & (df.E == e)
-        df.loc[mask, "FY"] = df[mask].squeeze().FY * pert
+        idx = df.query(f"ZAM=={zam} & ZAP=={zap} & MT=={mt} & E=={e}").index
+        fy_pert = df.loc[idx, :].set_index("ZAP").FY * pert
```

Comment:
If pert is a list, then you don't need to match the index anymore for the product and this could be simplified